service/matching/taskQueueManager.go

```diff
@@ -398,6 +402,15 @@ func (c *taskQueueManagerImpl) DispatchQueryTask(
 	return c.matcher.OfferQuery(ctx, task)
 }
 
+// TODO: Use signalIfFatal
```

Comment:
I think you only need to do that for the mutate call.. the read doesn't check the lease. anyway, are you planning to do it in this pr or later?

Reply:
I'll do it now